cdap-docs/developers-manual/source/data-exploration/streams.rst

```diff
@@ -0,0 +1,214 @@
+a format and schema of their own. For example, event bodies may be comma delimited
+text or Avro encoded binary data. In these cases it is possible to set a format and schema
+on a Stream, enabling more powerful queries.
+
+Let's take a closer look at attaching format and schema on streams.
+
+Stream Format
+-------------
+
+A format defines how the bytes in an event body can be read as some higher level object.
+For example, the CSV (comma separated values) format can read each value in comma delimited text 
+as a separate column of some given type. Each format has 
+a schema that describes the structure of data the format can read. Some formats, like the avro format,
+require a schema to be explicitly given. Other formats, like the CSV format, have a default schema. 
+
+Format is a configuration setting that can be set on a stream. This can be done through the
+RESTful API, or by using the CLI::
+
+  set stream format <stream-id> <format-name> [<schema>] [<settings>]
+  set stream format mystream csv "f1 int, f2 string"
+
+As mentioned above, a format may support different schemas, which will be discussed in more detail
+in the next section.
+
+It is important to note that formats are applied at read time.
+When events are added to a stream, there are no checks performed to enforce that
+all events added to the stream are readable by the format you have set on a stream.
+If any stream event cannot be read by the format you have set, your query will not work.
+
+Schema
+------
+CDAP schemas are adopted from Avro schemas `<http://avro.apache.org/docs/1.7.3/spec.html#schemas>`__
+with a few differences:
+ 
+  * Map keys do not have to be strings, but can be any type.
+  * No "name" property for enum type. 
+  * No support of "doc" and "aliases" in record and enum types.
+  * No support of "doc" and "default" in record field.
+  * No "fixed" type.
+
+There are a few additional limitations on the types of schemas that can be used for exploration: 
+ 
+  * Schemas must be a record of at least one field.
+  * Enums are not supported.
+  * Unions are not supported, unless it is a union of a null and another type, representing a nullable type.
+  * Recursive types are not supported. This means you cannot have a record field that references itself in one of its fields.
+
+Data exploration using Impala has additional limitations:
+
+  * Fields must be a scalar type (no maps, arrays, or records)
+
+On top of these general limitations, each format has its own restrictions on the types
+of schemas they support. For example, the CSV format does not support maps or records as
+data types. Please see the javadocs for each format for more information about the types
+of schemas they support. 
+
+Schema Syntax
+-------------
+Schemas are represented as JSON Objects, following the same format as Avro schemas
+`<http://avro.apache.org/docs/1.7.3/spec.html#schemas>`__. 
+The JSON representation is used by the RESTful APIs.
```

Comment:
The JSON representation is used by the HTTP RESTful APIs, while the CDAP CLI supports a SQL-like syntax.